features/collection/thresholding/impl/src/main/java/org/opennms/netmgt/threshd/AbstractThresholdEvaluatorState.java

```diff
@@ -128,6 +128,8 @@ public Long load(String key) {
 
     static abstract class AbstractState implements Serializable {
         String interpolatedExpression = null;
+        boolean cached = false;
```

Comment:
Yeah, this is needed since `thresholdValues` can exist before and after reload.  
Doesn't state itself get `reset` when reloading ? Looks like it's not. When merging thresholds, looks like it retains previous ThresholdStates and just loads the config again.  There are several tests with  `ThresholdingVisitor`  `reload` that mimics this  behavior.

Reply:
Not sure I follow, looks like very writer of `thresholdValues` also sets the cached field. Also upon deserializing an object with `thresholdValues` set wouldn't `cached` also be set in that case?

Reply:
`clearStateBeforePersist` unsets the cache which gets called in `ThresholdingVisitor` `reload`.  If you are saying that `thresholdValues` can be set as null or `Optional.empty` in `clearStateBeforePersist` that's an option but I chose this as this is common to both `interpolatedExpression` and `thresholdValues`.

Reply:
Ok, I missed the part where it was getting cleared in `clearStateBeforePersist`, I was thinking it was always being set in a tandem with the value. No worries, can leave as is.